drafting.typ

```diff
@@ -447,6 +451,7 @@
   box[
     #place(path(stroke: props.stroke, ..path-pts))
     #place(dx: dist-to-margin + 1 * pct, dy: dy, [#note-rect<margin-note>])
+    #_note-outline-entry(props, body)
```

Comment:
Hey all ... sorry to butt in here, but this causes a bit of a bug with stateful content.

I've been using the @ntjess repo because it's handling right/left based on inside/outside margins, so I will provide a MRE showing the problem.  

The issue is that line 453 and 454 **both** write `body` (line 453 has `body` as a parameter).  The problem is that  by writing out the same content WHEN that content has a stateful call to something, will result in that stateful code being called twice.  

My minimum reproducible example uses `#cite()` which maintains state about how many citations are made on a page ( the chicago style will write the bibliographic reference as a footnote to the bottom of the page)

So if do something like
```typ
#lorem(10);#margin-note([#lorem(10)  #cite(<codd_relational_1970>) #lorem(20)])#lorem(200)
``` 

I will see the footnote superscript in the margin note.  Good.
<img width="271" alt="image" src="https://github.com/user-attachments/assets/d29ccfaa-e02b-4ac7-80de-01088c7014cc" />

But the bottom of the page has two references, indicating that the hidden content has statefully
bumped up the citation count.

<img width="380" alt="image" src="https://github.com/user-attachments/assets/28a32e77-a179-4919-9ee3-5cbdb40d9e7d" />

I'm not exactly sure how to solve this, because I don't understand exactly the purpose of 
```typ
  #_note-outline-entry(props, body)
```

but I have commented that out for my particular needs (I am vendoring the file).

**Complete MRE:**
```typ
#import "typst-drafting/drafting.typ": *

#let outside_margin = 2.25in
#let inside_margin = 1in


#set page(
    paper: "a4",
    margin: (
      inside: inside_margin,
      outside: outside_margin,
  )
)
#set-page-properties()
#set-margin-note-defaults(stroke: none)


= This paragraph has a normal cite
#lorem(10);#cite(<codd_relational_1970>);#lorem(40)



#pagebreak()
= This paragraph contains a margin note with a citation that double-counts

#lorem(10);#margin-note([#lorem(10)  #cite(<codd_relational_1970>) #lorem(20)])#lorem(200)

#pagebreak()
#bibliography(style: "chicago-notes", "./data/mre.bib")


////////// mre.bib contents
// @article{codd_relational_1970,
// 	title = {A Relational Model of Data for Large Shared Data Banks},
// 	volume = {13},
// 	doi = {https://doi.org/10.1145/362384.362685},
// 	number = {6},
// 	author = {Codd, E F},
// 	date = {1970},
// 	langid = {english},
// 	keywords = {delightql-cite},
// 	file = {Codd - 1970 - A Relational Model of Data for Large Shared Data B.pdf:/Users/danieleklund/Zotero/storage/QM8TJ6B9/Codd - 1970 - A Relational Model of Data for Large Shared Data B.pdf:application/pdf},
// }

```